src/client/api/json_parser/common.c

```diff
@@ -62,21 +62,22 @@ int json_parser_common_address_deserialize(cJSON *json_obj, char const *const js
 /*
 {
   "type": 0,
-  "pubKeyHash": "21e26b38a3308d6262ae9921f46ac871457ef6813a38f6a2e77c947b1d79c942"
+  "pubKeyHash": "0x21e26b38a3308d6262ae9921f46ac871457ef6813a38f6a2e77c947b1d79c942"
 }
 or
 {
   "type": 8,
-  "aliasId": "a3308d6262ae9921f46aa3308d6262ae9921f46a"
+  "aliasId": "0xa3308d6262ae9921f46aa3308d6262ae9921f46a"
 }
 or
 {
   "type": 16,
-  "nftId": "a3308d6262ae9921f46aa3308d6262ae9921f46a"
+  "nftId": "0xa3308d6262ae9921f46aa3308d6262ae9921f46a"
 }
 */
 cJSON *json_parser_common_address_serialize(address_t *address) {
-  char addr_str[BIN_TO_HEX_STR_BYTES(ADDRESS_MAX_BYTES)] = {};
+  char addr_str[BIN_TO_HEX_STR_BYTES(ADDRESS_MAX_BYTES) + JSON_HEX_ENCODED_STRING_PREFIX_LEN] = {};
+  memcpy(addr_str, "0x", JSON_HEX_ENCODED_STRING_PREFIX_LEN);
```

Comment:
maybe support `prefix` parameter in `hex_2_bin` and `bin_2_hex`? so don't need to copy the prefix everywhere, like
```
int hex_2_bin(char const str[], size_t str_len, char const prefix[], byte_t bin[], size_t bin_len) 
int bin_2_hex(byte_t const bin[], size_t bin_len, char const prefix[], char str_buf[], size_t buf_len)
```
the prefix parameter is optional, set to NULL for regular conversion

```
bin_2_hex(address->address, ED25519_PUBKEY_BYTES, "0x", addr_str, sizeof(addr_str))
```
even define "0x" as `JSON_HEX_ENCODED_STRING_PREFIX`

Reply:
Thank you for this code improvement suggestion. It was done in this way.